### PR TITLE
Enhancement: Limit excessively long content length when computing suggestions

### PR DESCRIPTION
**Pull request by Merinorus**
## Proposed change

Limit excessively long content length (> 1 million characters) when computing suggestions. The original content is not affected.

This prevents excessive processing times on very large documents by limiting the text analyzed during date parsing, tag prediction, and correspondent matching.

Classifier training is not affected: its speed and accuracy remain unchanged. This setting is primarily useful for balancing suggestion speed and accuracy when navigating individual documents. It can be adjusted without retraining the classifier.

This is the 5th optimization mentioned in this thread: https://github.com/paperless-ngx/paperless-ngx/discussions/10156

Initially, I added a separate setting to adjust the head/tail ratio: https://github.com/paperless-ngx/paperless-ngx/commit/fcecaf1b9b39a642473c6873d800241f8d6a3b70. However, tests showed minimal benefit, so I simplified to a fixed 80/20 split between the start and end of the document.
This [study](https://arxiv.org/pdf/2104.07225) shows a modest improvement of the head+tail crop compared to head only. It really depends on the document collection, of course. I guessed that in many administrative documents, there is useful information at the end: name, signature, date, etc. And many users use Paperless-ngx for this application (again, a personal guess, I may be wrong!). That's why I included the tail.

## Type of change

- [ ] Bug fix: non-breaking change which fixes an issue.
- [x] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by shamoon** [comment]
Yea so my concern here is this is YAEV™ (yet another environment variable) that most people are not going to even think to use (one really needs to understand how paperless works to even understand it). I suppose it could be called out in the “low powered devices” section but where this actually seems more useful is ridiculously long documents.

Additionally, suggestions aren’t blocking when the document loads (intentionally) and FYI #10319 contains a change such that suggestions are only automatically checked for docs with inbox tags which I also think mitigates this.

So if we do go ahead with this, perhaps better to use a sane default that really just addresses insanely long docs. Btw, do you personally actually have documents that are 1M chars (as I said from the beginning, I don’t think most users do)?

**Comment by Merinorus** [comment]
Oh, I didn't know about the automatic suggestions modified by your PR, I'll look into that.

On the almost 3000 documents I have:
- 30 are > 100 k characters (~1%)
- 10 are > 200 k characters
- 3 are about 500k characters, which is the maximum

The only corner case I have is a multilingual  200K char document (e.g.: a manual in 6 languages I never use). The stemmer's cache doesn't work well for this. Otherwise, the [classifier's cache](https://github.com/paperless-ngx/paperless-ngx/pull/10363) makes it fast enough: 500k characters takes about twice (1.5s) the time of 10k characters (0.8s).

So, yeah... It's best to avoid this environment variable and use a hardcoded limit of 1M characters. It would definitely be more suited to avoid edge cases. I'll rework this one.

File: src/documents/tests/utils.py [hunk 1]
```diff
@@ -426,3 +428,9 @@ def send_progress(
             payload["data"].update(extra_args)
 
         self.payloads.append(payload)
+
+
+def pseudo_random_text(length=100, seed=42):
```
Comment (stumpylog) [inline src/documents/tests/utils.py]: What do you think about [faker.text](https://faker.readthedocs.io/en/stable/providers/faker.providers.lorem.html#faker.providers.lorem.Provider.text) instead?  It is an "approximate number of characters", so this would need to check and maybe add more.  Or perhaps double the amount, then limit

Comment (Merinorus) [reply src/documents/tests/utils.py]: I thought it wasn't in the project requirements. I can do it with `faker`, for sure.

Comment (stumpylog) [reply src/documents/tests/utils.py]: It comes with factory-boy, which I want to start using more anyway

Comment (Merinorus) [reply src/documents/tests/utils.py]: The test looks cleaner now, thank you!

**Review by stumpylog — APPROVED** [review]
I think this looks fine, but maybe the title and description should be updated now, to match what it really does?

**Comment by Merinorus** [comment]
Yes, I modified the title and content. Feel free to adjust.
